File: Java/NWD.java
import java.util.Scanner;

class Main 
{
    public static void main(String[] arg) 
    {
        Scanner input = new Scanner(System.in);
        int t = input.nextInt();
        for (int i = 0; i < t; i++) 
        {
            int a = input.nextInt();
            int b = input.nextInt();
            System.out.println(NWD(a, b));
        }
        input.close();
    }

    private static int NWD(int a, int b)
     {
        while (b != 0) 
        {
            int temp = b;
            b = a % b;
            a = temp;
        }
        return a;
    }
}

File: C++/NWD.cpp
#include <iostream>

int nwd(int a, int b) 
{
    while (b != 0) 
    {
        int temp = b;
        b = a % b;
        a = temp;
    }
    return a;
}

int main() 
{
    int t, a, b;
    std::cin >> t;
    for (int i = 0; i < t; i++) 
    {
        std::cin >> a >> b;
        std::cout << nwd(a, b) << '\n';
    }
    return 0;
}